sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/CustomShuffleReaderExec.scala

```diff
@@ -128,6 +104,27 @@ case class CustomShuffleReaderExec private(
     Map("numSkewedPartitions" -> metrics)
   }
 
+  private def sendPartitionDataSizeMetrics(
+      executionId: String,
+      partitionMetrics: SQLMetric): Unit = {
+    val mapStats = shuffleStage.get.mapStats.get.bytesByPartitionId
```

Comment:
Let's follow the previous code: https://github.com/apache/spark/pull/28175/files#diff-a42cafdbb5870e28c4e03df50ffc44f6L111

If `shuffleStage.get.mapStats.isEmpty`, we send the metric value as 0 only once.